types/types.go

```diff
@@ -182,11 +183,13 @@ func (node *NodeConfig) GenerateConfig(dst, templ string) error {
 	if err != nil {
 		return err
 	}
-	defer f.Close()
 
 	_, err = f.Write(dstBytes.Bytes())
+	if err != nil {
+		return err
+	}
 
-	return err
+	return f.Close()
```

Comment:
what is the reason of removing `defer`? If `f.Write` errors, but the file has already been opened, then it won't be closed without a defer.

Reply:
This was not my free will but Deepsource complaining.
Here is the reference: https://deepsource.io/gh/srl-labs/containerlab/run/7183e662-b504-4b99-a10c-3047bf5473f6/go/GO-S2307?listindex=0

Deepsource seems to only inspect files or sections of files touched in the related commits. Thats why this "issue" popped up although I did not even mean to touch this part in first place.
So my changes in this case only follow the Deepsource recommendations.